instances/widgets.treasury-factory.near/widget/pages/dashboard/Portfolio.jsx

```diff
@@ -653,22 +654,44 @@ useEffect(() => {
     const deserialized = deserializeLockupContract(
       new Uint8Array([...lockupState].map((c) => c.charCodeAt(0)))
     );
-    const lockupTimestamp =
-      deserialized.lockup_information.lockup_timestamp.toString();
-    const releaseDuration =
-      deserialized.lockup_information.release_duration.toString();
+    let lockupStartTimestamp = deserialized.lockup_information.lockup_timestamp
+      ? deserialized.lockup_information.lockup_timestamp.toString()
+      : null;
+    let releaseDuration = deserialized.lockup_information.release_duration
+      ? deserialized.lockup_information.release_duration.toString()
+      : null;
     const totalAllocated =
       deserialized.lockup_information.lockup_amount.toString();
     const locked = nearBalances.contractLocked;
+    const transfersTimestamp = deserialized.lockup_information
+      .transfers_information?.transfers_timestamp
+      ? deserialized.lockup_information.transfers_information?.transfers_timestamp.toString()
+      : null;
+
+    if (!lockupStartTimestamp && transfersTimestamp) {
+      lockupStartTimestamp = transfersTimestamp;
+    }
+    let lockupEndTimestamp = Big(lockupStartTimestamp ?? "0")
+      .plus(releaseDuration ?? "0")
+      .toFixed();
+    let cliffTimestamp = null;
+    if (deserialized.vesting_information?.schedule) {
+      lockupStartTimestamp =
+        deserialized.vesting_information?.schedule.start_timestamp.toString();
+      cliffTimestamp =
+        deserialized.vesting_information?.schedule.cliff_timestamp.toString();
+      lockupEndTimestamp =
+        deserialized.vesting_information?.schedule.end_timestamp.toString();
+    }
+
     setLockupTotalAllocated(formatNearAmount(totalAllocated));
     setLockupVested(
       formatNearAmount(Big(totalAllocated).minus(locked).toFixed())
     );
+    setLockupCliffDate(cliffTimestamp ? convertToDate(cliffTimestamp) : null);
```

Comment:
My concern is, as we saw from debugging the tests, that the displayed date depends on the user timezone. So users across the globe will not see the same date here. I believe this should be addressed, so that it displays the same date for everyone? For example using UTC timezone?